06-building-a-program-start-to-finish/writing-tests/output/test.js

```diff
@@ -0,0 +1,133 @@
+const InvertedIndex = require('../src/inverted-index.js');
+const books = require('./books');
+
+describe('Inverted index Suite', () => {
+   // Create an instance of the Index class
+  const newIndex = new InvertedIndex();
+  const emptyBook = [];
+  const sampleSentence = 'As &you can see here, y*ou, #@have ==defined *the function';
+  const multipleSearch = 'Coverage lines for you';
+  beforeEach(() => {
+    newIndex.createIndex('books', books);
+  });
+
+  describe('Class Inverted Index', () => {
+    it('should be a class', () => {
+      expect(newIndex instanceof InvertedIndex).toBe(true);
+      expect(newIndex instanceof Object).toBe(true);
+      expect(typeof (newIndex)).toBe('object');
+    });
+  });
+
+  describe('Tokenize String', () => {
+    it('should be available in class InvertedIndex', () => {
+      expect(InvertedIndex.tokenize).toBeDefined();
+    });
+    it('should return an array containing alphabets only', () => {
+      expect(InvertedIndex.tokenize(sampleSentence)).not.toContain('&');
+    });
+    it('should return an array containing alphabets only', () => {
+      expect(InvertedIndex.tokenize(sampleSentence)).not.toContain(/&#\*@=/);
+    });
+    it('should return an array containing alphabets only', () => {
+      expect(InvertedIndex.tokenize(sampleSentence)).toEqual(['as',
+        'can', 'defined', 'function', 'have', 'here', 'see', 'the',
+        'you', 'you']);
+    });
+    it('should return an array containing the correct number of words', () => {
+      expect(InvertedIndex.tokenize(sampleSentence).length).toBe(10);
+    });
+  });
+
+  describe('Unique Words', () => {
+    it('should be available in class InvertedIndex', () => {
+      expect(InvertedIndex.uniqueWords).toBeDefined();
+    });
+    it('should return an array of words without duplicates', () => {
+      expect(InvertedIndex.uniqueWords(sampleSentence).length).toBe(9);
+    });
+    it('should return an array of words without duplicates', () => {
+      expect(InvertedIndex.uniqueWords(sampleSentence)).toEqual(['as',
+        'can', 'defined', 'function', 'have', 'here', 'see', 'the',
+        'you']);
+    });
+  });
+
+  describe('Read book data', () => {
+    it('should read content of json and return false if file is empty', () => {
+      const isValid = InvertedIndex.readFileData(emptyBook);
+      expect(isValid).toBe(false);
+    });
+    it('should read content of json and return true if file matches', () => {
+      const isValid = InvertedIndex.readFileData(books);
+      expect(isValid).toBe(true);
+    });
+  });
+
+  describe('Populate Index', () => {
+    it('should have an Index created', () => {
+      expect(newIndex.index.books).toBeDefined();
+    });
+    it('should accurately map words to their document location', () => {
```

Comment:
I like that you're testing for come words in the index, I'd like to see you test that the index created is *`exactly`* what was required (all words).